### PR TITLE
Provide warning for DX heating coil when DefrostEIR performance curve name is blank and defrost method is reverse cycle

### PR DESCRIPTION
**Pull request by rraustad**
## Pull request overview

Correct test for DX heating coil to require defrost EIR performance curve if defrost strategy is Reverse Cycle. Corrects issue #5700. No diff's expected in example files.
### Work Checklist

Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
- [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
- [x] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
  - Defect: This pull request repairs a github defect issue.  The github issue should be referenced in the PR description
    ### Review Checklist
    This will not be exhaustively relevant to every PR.
- [x] Code style (parentheses padding, variable names)
- [x] Functional code review (it has to work!)
- [x] If defect, results of running current develop vs this branch should exhibit the fix
- [ ] CI status: all green or justified
- [ ] Performance: CI Linux results include performance check -- verify this
- [x] Unit Test(s)
- C++ checks:
  - [ ] Argument types
  - [ ] If any virtual classes, ensure virtual destructor included, other things
- IDD changes:
  - [ ] Verify naming conventions and styles, memos and notes and defaults
  - [ ] If transition, add rules to spreadsheet
  - [ ] If transition, add transition source
  - [ ] If transition, update idfs
- [ ] If new idf included, locally check the err file and other outputs
- [ ] Documentation changes in place
- [ ] ExpandObjects changes??
- [ ] If output changes, including tabular output structure, add to output rules file for interfaces

### CODE REVIEW

**Comment by rraustad** [comment]
Expecting all green CI

File: src/EnergyPlus/DXCoils.cc [hunk 1]
```diff
@@ -2209,12 +2209,12 @@ namespace DXCoils {
 
 			// Only required for reverse cycle heat pumps
 			DXCoil( DXCoilNum ).DefrostEIRFT = GetCurveIndex( Alphas( 10 ) ); // convert curve name to number
-			if ( SameString( Alphas( 11 ), "ReverseCycle" ) && SameString( Alphas( 12 ), "OnDemand" ) ) {
+			if ( SameString( Alphas( 11 ), "ReverseCycle" ) ) {
```
Comment (rraustad) [inline src/EnergyPlus/DXCoils.cc:2212]: I remember adding this way back when I thought the curve was only used for OnDemand control type. The multispeed DX heating coil only checks for ReverseCycle so no need to correct that object's GetInput.

File: tst/EnergyPlus/unit/DXCoils.unit.cc [hunk 2]
```diff
@@ -221,11 +221,6 @@ namespace EnergyPlus {
 		// 	"! <DX Heating Coil Standard Rating Information>, Component Type, Component Name, High Temperature Heating (net) Rating Capacity {W}, Low Temperature Heating (net) Rating Capacity {W}, HSPF {Btu/W-h}, Region Number",
 		// 	" DX Heating Coil Standard Rating Information, , DX Heating coil, 6414.3, 6414.3, 6.58, 4" } ) ) );
 
-		// Clean up
-		DXCoil.deallocate();
-		DXCoilNumericFields.deallocate();
-		PerfCurve.deallocate();
-
```
Comment (rraustad) [inline tst/EnergyPlus/unit/DXCoils.unit.cc:228]: cleaned up some unnecessary deallocates in unit tests since these are done in clear-state

**Comment by lgu1234** [comment]
The fix provides meaningful error message Unit test performed as expected.

As @rraustad mentioned that it is odd to make a fatal when no defrost curve is provided. It is better to allow default curve value at 1 to make the simulation continue and provide a warning message to let a user know, similar process as #5162.

**Comment by rraustad** [comment]
If I were to default this I would expect a parabola with peak near the freezing point and falling off on either side. I haven't seen any data on this topic.

**Comment by lgu1234** [comment]
@rraustad It is possible to allow users to input a constant curve with value = 1. If this fails, do we reject a constant input?

**Comment by rraustad** [comment]
Yes, the user can enter a curve with a constant value of 1. I just would not use that as a default since I think it would cause high defrost energy use and users would not know any different or would believe that since E+ provided the curve it was valid.

**Pull request merged**